adding the definition for experimental

jpvajda opened the pull request:
This PR is part of an initiative to better define our experimental release process.

jpvajda commented:
@prasek 👋

prasek commented on src/content/basics/resources/release-stages.md:
```diff
@@ -18,6 +19,10 @@ graph LR;
 
+### Experimental
+
+An experimental release is in very volatile, active development. The release is not feature-complete, and breaking API changes are possible between individual versions. An experimental release may not move to Alpha / Beta, and should be used cautiously and for testing purposes only. An experimental release may be temporary and at any point could be deprecated.
```
@jpvajda think we'll want to use experimental to describe specific features, but not the whole release of a project. As such experimental features can be in a alpha, beta, or GA release of a project and would graduate from experimental -> mainline separate from the open source project release lifecycle.

For example the Apollo Router as a project is GA, but it has experimental support for the Rhai scripting feature:
https://www.apollographql.com/docs/router/customizations/rhai

Given that suggest we create a new section for experimental features:
- open source release stages
- cloud release stages
- experimental features

however, the experimental features are only applicable to open source projects, so we may want something like this:
- Apollo open source projects
  - release stages
  - experimental features
- Apollo cloud
   - release stages

Let's think on this and circle back. 🙏 

/cc @StephenBarlow

jpvajda commented:
@prasek Let me know what you think of this change.

prasek commented on src/content/basics/resources/release-stages.md:
```diff
@@ -46,6 +46,10 @@ A deprecated release (or individual feature) is no longer officially supported b
 
+### Experimental features
+
+An experimental feature is in very volatile, active development. The feature is not feature-complete, and breaking API changes are possible between individual versions. An experimental feature can exist in any of the release stages defined above and should be used cautiously and for testing purposes only. An experimental feature may be temporary and at any point could be deprecated.
```
```suggestion
An experimental feature is in very volatile, active development. The feature is not feature-complete, and breaking API changes are possible between individual versions. An experimental feature can exist in any of the release stages defined above and should be used cautiously and for testing purposes only. An experimental feature may be temporary and removed at any point without notice.
```

Would give more affordance for removing experimental features without notice instead of deprecated  which is usually reserved for non-experimental GA features. It's not saying we'll always remove experimental features without notice, just that we reserve the right to do so, which is one reason you should use them in non-test environment.

jpvajda commented:
@prasek nice suggestion! I merged it in. So we can leave this PR as "don't merge" until we do a final review of the internal process.

jpvajda commented:
`update` This is still under review.

jpvajda commented:
@prasek a few final edits for you to review 👀

prasek commented:
can we bump experimental to H2?

<img width="1872" alt="image" src="https://user-images.githubusercontent.com/284840/184990459-eef0cfd1-1c2d-4b18-ac5e-34520b655154.png">

StephenBarlow commented:
Ah yes I think we should also move it to the bottom since it could apply to either release pattern

StephenBarlow reviewed (approved):
Approved if the suggestions look good to implement!

jpvajda commented:
@prasek / @StephenBarlow great suggestions. I updated the location, made it an H2 header and added some additional introduction context per @debergalis 's feedback.

prasek reviewed (approved):
lgtm, thx!
